index.js

```diff
@@ -5,15 +5,18 @@ const fp = require('fastify-plugin')
 const Next = require('next')
 
 function fastifyNext (fastify, options, next) {
-  const { underPressure, noServeAssets, ...nextOptions } = options
-
+  const { underPressure, noServeAssets, hostname, port, ...nextOptions } = options
   if (underPressure) {
     const opts = typeof underPressure === 'object' ? underPressure : Object.create(null)
 
     fastify.register(require('@fastify/under-pressure'), opts)
   }
 
-  const app = Next(Object.assign({}, { dev: process.env.NODE_ENV !== 'production' }, nextOptions))
+  const app = Next(Object.assign({}, {
+    dev: process.env.NODE_ENV !== 'production',
+    hostname: hostname !== undefined ? hostname : '0.0.0.0',
```

Comment:
I don't understand. Why you need to pass `hostname` and `port` to `Next`?

Reply:
Really not a good idea to use `0.0.0.0` by default.

Reply:
Hey, because for obscure reason Next.js need pass port and hostname for prevent the undefined result 
According to the docs :
```js
// when using middleware `hostname` and `port` must be provided below
const app = next({ dev, hostname, port })
```
https://nextjs.org/docs/advanced-features/custom-server

this problem exist too with firebase (without fastfy server ) : https://github.com/vercel/next.js/discussions/41210

Reply:
> Really not a good idea to use `0.0.0.0` by default.

i can replace by **localhost** if you want

Reply:
Yes, please